src/graph/Graph.java

```diff
@@ -0,0 +1,123 @@
+package graph;
+
+import java.util.*;
+
+
+public abstract class Graph {
+
+  private Map<Integer, Vertex> vertices;
+
+  public Graph() {
+    this.vertices = new HashMap<>();
+  }
+
+  public Vertex getNode(int id) {
+    return this.vertices.get(id);
+  }
+
+  public Set<Integer> getNodesID() {
+    return vertices.keySet();
+  }
+
+  public Collection<Vertex> getNodes() {
+    return vertices.values();
+  }
+
+  public Vertex addNode(Vertex v) {
+    if (this.contains(v)) {
+      return this.getNode(v.getId());
+    }
+    this.vertices.put(v.getId(), v);
+    return v;
+  }
+
+  public Vertex addNode(int id) {
+    return this.addNode(new Vertex(id));
+  }
+
+  public Vertex addNode() {
+    if (this.isEmpty()) {
+      return this.addNode(0);
+    }
+    int maxId = Collections.max(this.vertices.keySet());
+    return this.addNode(maxId + 1);
+  }
+
+  public Vertex removeNode(int id) {
+    return this.vertices.remove(id);
+  }
+
+  public Vertex removeNode(Vertex v) {
```

Comment:
This is not very useful. While removing, we do not need the attribute, so passing a vertex object would not help too much.